cmd/rpc/server.go

```diff
@@ -120,8 +119,11 @@ func (s *Server) startRPC(router *httprouter.Router, port string) {
 	// Start RPC server
 	s.logger.Infof("Starting RPC server at 0.0.0.0:%s", port)
 	s.logger.Fatal((&http.Server{
-		Addr:    colon + port,
-		Handler: cor.Handler(http.TimeoutHandler(router, timeout, lib.ErrServerTimeout().Error())),
+		Addr:              colon + port,
+		ReadHeaderTimeout: timeout,
```

Comment:
having the same timeout for all operations may not be ideal if for example servers read heavier data than what they write, in practice it should be pretty similar but just as a possible suggestion the http timeouts in config could be separated as `httpReadTimeoutMS` and `httpWriteTimeoutMS` (the `ms` is to follow the same convention as the other timeouts in the config)